extensions/pkg/controller/infrastructure/actuator.go

```diff
@@ -13,16 +13,61 @@ import (
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
 )
 
-// Actuator acts upon Infrastructure resources.
+// Actuator acts upon [extensionsv1alpha1.Infrastructure] resources.
 type Actuator interface {
-	// Reconcile the Infrastructure config.
+	// Reconcile reconciles the [extensionsv1alpha1.Infrastructure] resource.
+	//
+	// Implementations should ensure that required infrastructure resources
+	// are created or updated to reach the desired state.
 	Reconcile(context.Context, logr.Logger, *extensionsv1alpha1.Infrastructure, *extensionscontroller.Cluster) error
-	// Delete the Infrastructure config.
+
+	// Delete is invoked when the [extensionsv1alpha1.Infrastructure]
+	// resource is deleted.
+	//
+	// Implementations should take care of cleaning up any infrastructure
+	// related resources (e.g. custom routes or subnets) which were created
```

Comment:
```suggestion
	// related resources (e.g. subnets, routers, etc.) which were created
```

My previous comment about `custom routes or subnets` was only related to the wait part during the force deletion. In general, extensions should not delete any resources it does not manage, which is more or less motivated by saying "custom routes".